modules/core/include/visp3/core/vpImageTools.h

```diff
@@ -409,6 +420,168 @@ void vpImageTools::crop(const unsigned char *bitmap, unsigned int width, unsigne
+    neighbour interpolation.
+
+  \param I : Input image to be transformed. 
+  \param T : 3x3 matrix corresponding to \f${\bf T}\f$ that contains all possible geometric
+  transformations, also called affine mapping.
+  \param Iwarp : Resulting warped image. 
+
+  An example that uses warpAffine() is given in testImageWarpAffine.cpp
+*/
+template<class Type>
+void vpImageTools::warpAffine(const vpImage<Type> &I,
+                              const vpMatrix &T,
+                              vpImage<Type> &Iwarp)
+{
+
+  vpMatrix transcp(3,5);
+  transcp[0][0] = 1; transcp[0][1] = 1; transcp[0][2] = 1;            transcp[0][3] = I.getWidth(); transcp[0][4] = I.getWidth();
+  transcp[1][0] = 1; transcp[1][1] = 1; transcp[1][2] = I.getWidth(); transcp[1][3] = I.getWidth(); transcp[1][4] = 1;
+  transcp[2][0] = 1; transcp[2][1] = 1; transcp[2][2] = 1;            transcp[2][3] = 1;            transcp[2][4] = 1;
+
+  transcp = T * transcp;
+
+  float max_1 = std::max(transcp[0][0], std::max(transcp[0][1], std::max(transcp[0][2], std::max(transcp[0][3], transcp[0][4])))); ///< max(transcp(0,:)) 
+  float min_1 = std::min(transcp[0][0], std::min(transcp[0][1], std::min(transcp[0][2], std::min(transcp[0][3], transcp[0][4])))); ///< min(transcp(0,:)) 
+
+  float max_2 = std::max(transcp[1][0], std::max(transcp[1][1], std::max(transcp[1][2], std::max(transcp[1][3], transcp[1][4])))); ///< max(transcp(1,:)) 
+  float min_2 = std::min(transcp[1][0], std::min(transcp[1][1], std::min(transcp[1][2], std::min(transcp[1][3], transcp[1][4])))); ///< min(transcp(1,:)) 
+
+  vpMatrix imgy(floor(max_1 - min_1 + 1), floor(max_2 - min_2 + 1));
+  vpMatrix imgx(floor(max_2 - min_2 + 1), floor(max_1 - min_1 + 1));
+
+  // meshgrid(min_2:max_2,min_1:min_1)
+  for (unsigned int i = 0; i < imgy.getRows(); i++) {
+    for (unsigned int j = 0; j < imgy.getCols(); j++) {
+      imgy[i][j] = min_2 + j;
+    }
+  }
+
+  // meshgrid(min_1:max_1,min_2:min_2)
+  for (unsigned int i = 0; i < imgx.getRows(); i++) {
+    for (unsigned int j = 0; j < imgx.getCols(); j++) {
+      imgx[i][j] = min_1 + j;
+    }
+  }
+
+  imgx = imgx.t();
+
+  //Backwards Transform (inverse mapping)
+
+  // Initialize Inverse Mapping matrix
+  vpMatrix X(3, imgx.getRows()*imgx.getCols());
+
+  vpColVector B(3);
+
+  // X = [ imgx(:) , imgy(:) , ones(imgx.getRows() * imgx.getCols(), 1) ]'
+  for (unsigned int i = 0; i < X.getRows(); i++) {
+    for (unsigned int j = 0; j < X.getCols(); j++) {
+      if (i == 0)  X[i][j] = imgx[j % imgx.getRows()][j / imgx.getRows()];
+      if (i == 1)  X[i][j] = imgy[j % imgy.getRows()][j / imgy.getRows()];
+      if (i == 2)  X[i][j] = 1;
+    }
+  }
+
+  // perform T \ X
+  for (unsigned int j = 0; j < X.getCols(); j++) {
+    B[0] = X[0][j];
+    B[1] = X[1][j];
+    B[2] = X[2][j];
+    T.solveBySVD(B, B); // Converted Columns of X to a ColVector for Linear Solver
+    X[0][j] = B[0];
+    X[1][j] = B[1];
+    X[2][j] = B[2];
+  }
+
```

Comment:
Not efficient enough. Instead of doing the SVD for each column, we could compute T inverse just one time and then obtain X = T^-1 * B

The code becomes something like

```
  vpMatrix Tinv = T.inverseByLU();

  // perform T \ X
  for (unsigned int j = 0; j < X.getCols(); j++) {
    B[0] = X[0][j];
    B[1] = X[1][j];
    B[2] = X[2][j];
    vpMatrix X_ = Tinv * B;

    X[0][j] = X_[0][0];
    X[1][j] = X_[1][0];
    X[2][j] = X_[2][0];
  }
```